h2o-algos/src/main/java/hex/ensemble/MetalearnerProvider.java

```diff
@@ -0,0 +1,12 @@
+package hex.ensemble;
+
+import hex.Model.Parameters;
+
+public interface MetalearnerProvider<M extends Metalearner, P extends Parameters> {
+
+    String getName();
+
+    P newParameters();
```

Comment:
You are using `ModelBuilder.make` to create the metalearner instance (eg. GLM), this method also automatically supplies a new (and modifiable) instance of the default parameter object. Can we simplify the API and remove the need to have `newParameters`?

Reply:
@michalkurka this would not work for "AUTO" for example or any additional metalearner that would not be named exactly like its algo: we can imagine having more than one metalearner per algo (with different default params).

This `newParameters` method is called by `StackedEnsembleParameters`, so at a time where we don't havecreated the metalearner instance yet...

I could also suggest to keep the `newParameters` but make it a default interface method:
```
default P newParameters() {
   return newInstance().createBuilder()._parms;
}
```
not sure it is very efficient, nor elegant to have this in an interface though.
wdyt?

Reply:
> @michalkurka this would not work for "AUTO" for example or any additional metalearner that would not be named exactly like its algo: we can imagine having more than one metalearner per algo (with different default params).

My thought was to keep the `newInstance` method (for "AUTO"...) but remove `newParameters`, `newInstance` already provides the parameters. In the returned metalearner.

> This newParameters method is called by StackedEnsembleParameters, so at a time where we don't have created the metalearner instance yet...

https://github.com/h2oai/h2o-3/pull/4273/files#diff-91d255cd89e95ed4a240332df7d9002bR82

IMHO a special case, you could address by calling `newInstance().createBuilder()._parms` there.

I am also fine with keeping as-is.

Reply:
sold! less is more..

Reply:
@michalkurka  waiting for the build results and then merging